module/dkg.go

```diff
@@ -0,0 +1,42 @@
+package module
+
+import "github.com/onflow/flow-go/model/messages"
+
+// DKGContractClient enables interacting with the DKG smart contract. This
+// contract is deployed to the service account as part of a collection of
+// smart contracts that facilitate and manage epoch transitions.
+//
+// TODO type for DKGResult? - should be all participant public keys plus group public key
+type DKGContractClient interface {
+
+	// Broadcast broadcasts a message to all other nodes participating in the
+	// DKG. The message is broadcast by submitting a transaction to the DKG
+	// smart contract. An error is returned if the transaction has failed and
+	// should be re-submitted.
+	Broadcast(msg messages.DKGMessage) error
+
+	// ReadBroadcast reads the broadcast messages from the smart contract for
+	// a particular phase. All messages for the given phase that have been
+	// successfully broadcast will be returned, regardless of whether those
+	// messages have already been read by a previous call to ReadBroadcast.
+	//
+	// Messages are returned in the order in which they were broadcast (received
+	// and stored in the smart contract).
+	//
+	// DKG nodes should call ReadBroadcast one final time once they have
+	// observed the phase deadline trigger to guarantee they receive all
+	// messages for that phase.
+	//
+	// OPTIONAL: add a messageIndex parameter to only receive messages we have
+	// not already received in previous calls
+	ReadBroadcast(epochCounter uint64, phase messages.DKGPhase) ([]messages.DKGMessage, error)
```

Comment:
```suggestion
	ReadBroadcast(fromIndex uint, referenceBlock flow.Identifier) ([]messages.DKGMessage, error)
```

From discussion today we can change this to omit the phase/counter. We will need to be able to specify a reference point block for the query to do the DKG phase transition. We discussed using a message index as well, for which there is a transaction to support [here](https://github.com/onflow/flow-core-contracts/pull/108/files#diff-995b8991446500a0f054ed4717e52d26475c3e090e3a7e15eaf5fdd4f537a6d6R3)